torchdynamo/convert_frame.py

```diff
@@ -101,6 +103,52 @@ def _fn(*args, **kwargs):
     return _fn
 
 
+def has_tensor_in_frame(frame):
+    """Check if the frame has torch.* related bits"""
+    # Check if the function was decorated with torchdynamo.optimize
+    if frame.f_code in always_optimize_code_objects:
+        return True
+
+    # Check if there is global import of torch.*
+    for co_name in frame.f_code.co_names:
+        if co_name in frame.f_globals:
+            if is_allowed(frame.f_globals[co_name]):
+                return True
+
+    def has_tensor(obj):
```

Comment:
I think this will infinite loop if given:
```
x = []
x.append(x)
```

Memoize and keep track of all `id(obj)` seen to prevent that